src/test/java/org/opensearch/security/transport/SecurityInterceptorTests.java

```diff
@@ -249,17 +252,16 @@ final void completableRequestDecorate(
         TransportResponseHandler handler,
         DiscoveryNode localNode
     ) {
+        securityInterceptor.sendRequestDecorate(sender, connection, action, request, options, handler, localNode);
+        verifyOriginalContext(user);
+        try {
+            senderLatch.get().await(1, TimeUnit.SECONDS);
```

Comment:
This waits around for either sender to complete its validation - previously those assertions were ignored